### ISSUE TITLE
PHP Fatal error:  Class 'PHPCI\\\\Controller' not found 

### ISSUE THREAD

**Issue by hghesteemed** [Block8/PHPCI#1023]
New install of PHPCI and builds cannot run. I have updated the installation and tried to debug the routing like in a previous case. I get this in my web server error log

PHP Fatal error:  Class 'PHPCI\\\\Controller' not found in /var/www/html/phpci/vendor/block8/b8framework/b8/Application.php on line 93
How do i fix it?

**Comment by axyr** [comment]
+1 on osx yosemite

**Comment by tvidovic** [comment]
got the same issue.
PHP Fatal error:  Class 'PHPCI\\\\Controller' not found in /phpci/vendor/block8/b8framework/b8/Application.php on line 93

**Comment by mlewis-everley** [comment]
I also get this issue on Apache (haven't tried on nginx yet). I am sure this is a duplicate  another issue that has been logged, but I cannot actually find the original now...

**Comment by icyfire** [comment]
got the same issue on nginx too.

**Comment by sclearion** [comment]
Is this gonna be fixed

**Comment by adamazing** [comment]
@mlewis-everley I went looking and found the following, possibly duplicate/related issues:
#642 
#410 
#246

**Comment by sclearion** [comment]
Seen those. None of them provide a solution. Im on the latest version

**Comment by adamazing** [comment]
The common thread through those issues seems to be "routing issues" which could be caused by a number of factors. Can you share your apache/nginx config and folder layout? Is PHPCI in a subfolder or subdomain? Platform? Versions?

**Comment by mlewis-everley** [comment]
Cannot give config at the moment, will have a look on Monday (when I am
back in work).

All issues for me have been using Apache 2.2 and 2.4, on CentOS 6.5 and
Ubuntu 12.

All times I was using PHP 5.4.x.

Have tried both as a sub directory and as a sub domain (phpci.xxx.co.uk).

Hope that helps a little...

Cheers,

Mo

**Comment by Adirelle** [comment]
See https://github.com/Block8/PHPCI/pull/976.

**Comment by GregJPreece** [comment]
I am also having this issue, on Ubuntu Server 14.04, running Lighttpd and PHP 5.5.9. PHPCI's front-end is running, and builds seem to trigger my Phing build file, leave logs behind, etc. Builds show up as completed in the front-end, but in the back they have not been copied to their final destination. My error message is identical to the OP here.

#410 seems to imply that this is a routing issue of some kind. I have cloned the repo and will take a look myself if the software owner doesn't beat me to it.

**Comment by mpeshev** [comment]
Confirming that the issue still exists, Ubuntu 14.04, PHP 5.5.9, Apache. Build seems to be created, but no stats or data are visible and it's triggered as "pending".

**Comment by klaudi** [comment]
Have anybody fixed that?

**Comment by Caroga** [comment]
I don't know if this is anyhow related to the subject but as I find that in most of these cases the error is thrown as a result of routing issues, I will just post my nginx configuration below as this solved the same error in my case:
```
server {
    listen 80;
    server_name ci.server.tld;
    root /var/www/phpci/public;
    index index.php;

    location / {
        try_files $uri $uri/ /index.php?$args;
    }

    location ~ [^/]\.php(/|$) {
        fastcgi_split_path_info ^(.+?\.php)(/.*)$;
        if (!-f $document_root$fastcgi_script_name) {
            return 404;
        }
        fastcgi_index index.php;
        include fastcgi_params;
        fastcgi_pass unix:/var/run/php5-fpm.sock;
    }
}
```

**Comment by hghesteemed** [comment]
Closed as it seems like nobody maintains this. Moved on to other solutions.

**Comment by dancryer** [comment]
So there are two separate problems here, one is that builds are not running and the other is that there's an error coming up in the error log. 

**Problem 1 - PHP Fatal error: Class 'PHPCI\\Controller' not found in phpci/vendor/block8/b8framework/b8/Application.php on line 93**

Whilst annoying, this is effectively a non-issue. This comes up on English language installations as there's a file included that does not exist (/assets/js/plugins/datepicker/locales/bootstrap-datepicker.en.js) This has been fixed for v1.7, the patch can be found here: https://github.com/Block8/PHPCI/commit/c474a5c73a388b93a4daad1d9fcce03fc664f4c1#diff-4c3dcb804d8e696bfa0db20c8e5c6860

**Problem 2 - Builds not running**

In order for builds to run, you need to set up PHPCI to run them. In v1.6 (the current stable release) and below, there are two ways to do this:

1. Run PHPCI as a daemon: https://www.phptesting.org/wiki/Run-Builds-Using-a-Daemon - This continuously checks the database for new builds and executes them as and when they become available.

2. Run PHPCI as a cron-job: https://www.phptesting.org/wiki/Run-Builds-Using-Cron - This will execute all pending builds when it is run, and then exit.

In v1.7, coming soon - there's also the option to run PHPCI as a worker against a Beanstalkd queue. This will be the preferred method going forward. This will be documented in due course. 

Hope this helps, please open a new issue if the above does not solve your specific problems.
